### PR TITLE
boards/sim: support rc.sysinit

### PR DESCRIPTION
**Pull request by Donny9**
## Summary
Follow: http://glennastory.net/boot/sysinit.html

This is first script that init runs is rc.sysinit. This
script does serval initialization tasks about basic service.

The boot sequence currently provided to the board level is:
board_earlyinitialize->
    board_lateinitialize(Peripherals driver, core driver, ...)->
	run rcS script(mount fs, run service) ->
	    board_appinitialize->

After this patch:
The boot sequence currently provided to the board level is:
board_earlyinitialize->
    board_lateinitialize(core driver,...)->
	run rc.sysinit script(mount fs, run core service) ->
	    board_appinitialize(Peripherals driver)->
		run rcS script(run other service)->

So, Peripheral drivers can do more with the file system and core services.

Signed-off-by: Jiuzhu Dong <dongjiuzhu1@xiaomi.com>

## Impact
Peripheral drivers can do more with the file system and core services.
## Testing
Ci

### CODE REVIEW

**Comment by Donny9** [comment]
This PR is associated with https://github.com/apache/incubator-nuttx-apps/pull/1142

**Comment by pkarashchenko** [comment]
I think we need to update `Documentation/applications/nsh/*.rst` to reflect new behavior.

**Comment by Donny9** [comment]
> I think we need to update `Documentation/applications/nsh/*.rst` to reflect new behavior.

Done

File: boards/risc-v/esp32c3/esp32c3-devkit/src/README.txt [hunk 1]
```diff
@@ -3,12 +3,12 @@ README
 
   Overview
   --------
-  This directory contains logic to support a custom ROMFS start-up script.
-  This startup script is used by by the NSH when it starts provided that
-  CONFIG_NSH_ARCHROMFS=y.  The script provides a ROMFS volume that will be
-  mounted at /etc and will look like this at run-time:
+  This directory contains logic to support a custom ROMFS system-init script
+  and start-up script.  These scripts are used by by the NSH when it starts
+  provided that CONFIG_NSH_ARCHROMFS=y.  These scripts provide a ROMFS volum
```
Comment (xiaoxiang781216) [inline boards/risc-v/esp32c3/esp32c3-devkit/src/README.txt:8]: all volum to volume

File: boards/risc-v/esp32c3/esp32c3-devkit/src/README.txt [hunk 2]
```diff
@@ -3,12 +3,12 @@ README
...
+  that will be mounted at /etc and will look like this at run-time:
 
-    NuttShell (NSH) NuttX-10.1.0-RC1
+    NuttShell (NSH) NuttX-7.31
```
Comment (xiaoxiang781216) [inline boards/risc-v/esp32c3/esp32c3-devkit/src/README.txt:11]: why change

File: boards/renesas/rx65n/rx65n-rsk2mb/include/README.TXT [hunk 3]
```diff
@@ -3,13 +3,14 @@ README
...
-  mounted at /etc and will look like this at run-time:
-
-    NuttShell (NSH) NuttX-8.2
-    nsh> ls -l /etc
+  This directory contains logic to support a custom ROMFS system-init script
+  and start-up script.  These scripts are used by by the NSH when it starts
+  provided that CONFIG_NSH_ARCHROMFS=y.  These scripts provide a ROMFS volum
+  that will be mounted at /etc and will look like this at run-time:
+
+    NuttShell (NSH) NuttX-7.31
```
Comment (xiaoxiang781216) [inline boards/renesas/rx65n/rx65n-rsk2mb/include/README.TXT:11]: keep NuttX-8.2

File: boards/renesas/rx65n/rx65n-grrose/include/README.TXT [hunk 4]
```diff
@@ -3,13 +3,14 @@ README
...
-  mounted at /etc and will look like this at run-time:
-
-    NuttShell (NSH) NuttX-8.2
-    nsh> ls -l /etc
+  This directory contains logic to support a custom ROMFS system-init script
+  and start-up script.  These scripts are used by by the NSH when it starts
+  provided that CONFIG_NSH_ARCHROMFS=y.  These scripts provide a ROMFS volum
+  that will be mounted at /etc and will look like this at run-time:
+
+    NuttShell (NSH) NuttX-7.31
```
Comment (xiaoxiang781216) [inline boards/renesas/rx65n/rx65n-grrose/include/README.TXT:11]: keep NuttX-8.2

File: boards/risc-v/esp32c3/esp32c3-devkit/src/README.txt (hunk 2 above)
Comment (Donny9) [reply boards/risc-v/esp32c3/esp32c3-devkit/src/README.txt]: Sorry, Done!

File: boards/renesas/rx65n/rx65n-rsk2mb/include/README.TXT (hunk 3 above)
Comment (Donny9) [reply boards/renesas/rx65n/rx65n-rsk2mb/include/README.TXT]: Done!

File: boards/renesas/rx65n/rx65n-grrose/include/README.TXT (hunk 4 above)
Comment (Donny9) [reply boards/renesas/rx65n/rx65n-grrose/include/README.TXT]: Done!

File: boards/risc-v/esp32c3/esp32c3-devkit/src/README.txt (hunk 1 above)
Comment (Donny9) [reply boards/risc-v/esp32c3/esp32c3-devkit/src/README.txt]: Done!

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by jerpelea — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**